### PR TITLE
Add support for Symfony 6

### PR DESCRIPTION
**Pull request by loic425**

### CODE REVIEW

File: tests/Core/User/AdvancedUserInterfaceTest.php [hunk 1]
```diff
@@ -50,9 +50,6 @@ static function (\ReflectionMethod $reflectionMethod): string {
         Assert::assertContains('isCredentialsNonExpired', $declaredMethods);
         Assert::assertContains('isEnabled', $declaredMethods);
         Assert::assertContains('getRoles', $declaredMethods);
-        Assert::assertContains('getPassword', $declaredMethods);
```
Comment (loic425) [inline tests/Core/User/AdvancedUserInterfaceTest.php]: @lchrusciel @Zales0123 Maybe we have keep these 3 getters or maybe move them on Sylius User interface instead.

Comment (loic425) [reply tests/Core/User/AdvancedUserInterfaceTest.php]: I added a bc-layer directly on this package

File: tests/Core/User/AdvancedUserInterfaceTest.php [hunk 2]
```diff
@@ -52,7 +52,7 @@ static function (\ReflectionMethod $reflectionMethod): string {
         Assert::assertContains('getRoles', $declaredMethods);
         Assert::assertContains('getPassword', $declaredMethods);
         Assert::assertContains('getSalt', $declaredMethods);
-        Assert::assertContains('getUsername', $declaredMethods);
+        Assert::assertContains('getSalt', $declaredMethods);
```
Comment (Zales0123) [inline tests/Core/User/AdvancedUserInterfaceTest.php]: I believe it should be reverted 🖖

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, @loic425! 🏅

**Pull request merged**